### PR TITLE
Feat consumer subscription portal

### PR DESCRIPTION
**Pull request by bali710**
it has included a profile page Front end and Backend APIs
customer can edit info and see the List of Thrifts mark them done and delete as well

### CODE REVIEW

File: components/consumer/profile/Thrifts.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
+import { useState, useEffect } from "react";
+import HeaderComponent from "@/components/utility/Header";
+import SubscriptionModal from "@/components/scoped/SubscriptionModal";
+import ButtonComponent from "@/components/utility/Button";
+// import { useUser } from "@/context/UserContext";
```
Comment (pauloouriques) [inline components/consumer/profile/Thrifts.js]: remove this

File: components/consumer/profile/Thrifts.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+          id: id,
+        }),
+      });
+
+      if (response.ok) {
+        const resp = await response.json();
+        const index = consumerData.ThriftList.findIndex(
+          (elem) => elem.id === id
+        );
+        console.log("index", index);
```
Comment (pauloouriques) [inline components/consumer/profile/Thrifts.js]: remove this

File: pages/consumer/profile/index.js [hunk 3]
```diff
@@ -4,248 +4,68 @@ import SubscriptionModal from "@/components/scoped/SubscriptionModal";
 import ButtonComponent from "@/components/utility/Button";
 // import { useUser } from "@/context/UserContext";
 import { NotificationManager } from "react-notifications";
-
 import { signOut } from "next-auth/react";
 import { Loading } from "@nextui-org/react";
+import { QRCode } from "react-qrcode-logo";
+import debounce from "lodash.debounce";
+import ConsumerInfo from "@/components/consumer/profile/ConsumerInfo";
+// import ThriftList from "../../../components/consumer/profile/thriftList";
```
Comment (pauloouriques) [inline pages/consumer/profile/index.js]: remove

File: pages/consumer/profile/index.js [hunk 4]
```diff
@@ -4,248 +4,68 @@ import SubscriptionModal from "@/components/scoped/SubscriptionModal";
...
-      return;
-    } else if (!isValidEmail(user.email)) {
-      NotificationManager.error("Invalid email!");
-      return;
-    }
-    setUpdating(true);
-
-    setUpdating(false);
-
-    // try {
```
Comment (pauloouriques) [inline pages/consumer/profile/index.js:39]: remove

File: pages/consumer/profile/indexBackup.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
+import { useState, useEffect } from "react";
+import HeaderComponent from "@/components/utility/Header";
+import SubscriptionModal from "@/components/scoped/SubscriptionModal";
+import ButtonComponent from "@/components/utility/Button";
+// import { useUser } from "@/context/UserContext";
```
Comment (pauloouriques) [inline pages/consumer/profile/indexBackup.js]: remove

File: pages/consumer/profile/indexBackup.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+      return;
+    } else if (!isValidEmail(user.email)) {
+      NotificationManager.error("Invalid email!");
+      return;
+    }
+    setUpdating(true);
+
+    setUpdating(false);
+
+    // try {
```
Comment (pauloouriques) [inline pages/consumer/profile/indexBackup.js]: remove

**Review by pauloouriques — CHANGES_REQUESTED** [review]
clean up needed and also make sure to add prettier to your IDE

File: components/consumer/profile/Thrifts.js (hunk 1 above)
Comment (bali710) [reply components/consumer/profile/Thrifts.js]: removed the unnecessary imports and commented code from here

File: components/consumer/profile/Thrifts.js (hunk 2 above)
Comment (bali710) [reply components/consumer/profile/Thrifts.js]: removed the unnecessary console from this file

File: pages/consumer/profile/index.js (hunk 4 above)
Comment (bali710) [reply pages/consumer/profile/index.js:39]: console removed

File: pages/consumer/profile/indexBackup.js (hunk 6 above)
Comment (bali710) [reply pages/consumer/profile/indexBackup.js]: this file is no longer required deleted

**Review by pauloouriques — APPROVED** [review]
lgtm

**Pull request merged**